cpu/stm32/Kconfig

```diff
@@ -36,8 +36,8 @@ config ERROR_MODULES_CONFLICT
     default "On STM32, the RTC and RTT map to the same hardware peripheral." if MODULE_PERIPH_RTC && MODULE_PERIPH_RTT
     depends on CPU_STM32
 
-orsource "kconfigs/*/Kconfig"
-orsource "kconfigs/*/Kconfig.lines"
-orsource "kconfigs/*/Kconfig.models"
+orsource "kconfigs/$(CPU_FAM)/Kconfig"
+orsource "kconfigs/$(CPU_FAM)/Kconfig.lines"
+orsource "kconfigs/$(CPU_FAM)/Kconfig.models"
```

Comment:
This is required because of name clashing between the symbols in the different `Kconfig.clock` right?

A potential problem that I see, is that by introducing more CPU/board-dependent environmental variables to the include paths, the Kconfig tree diverges more and more among boards. This may get harder to follow and to inspect the whole tree at ones (e.g. for building doc or other analysis). I'm actually working on a branch to see if it could be possible to get rid of `$(BOARD)` and `$(CPU)` in the main Kconfig file, and handle the board selection via `board.config` files.

Do you think adding prefixes to the symbols would be an alternative?

Reply:
> This is required because of name clashing between the symbols in the different Kconfig.clock right?

Yes exactly.

And you are right that CPU_FAM might be removed when we switch to Kconfig only configuration management. So the solution proposed by this PR is not ideal.

> Do you think adding prefixes to the symbols would be an alternative?

That would work. The main issue I see is that setting a custom value from command line (using CFLAGS) will be less consistent between the different groups of family even if they are about the same thing (using HSE as SYSCLK or setting a custom PLL parameter).

What do you think of the following (only include the clock config if using the corresponding family) ?

```kconfig
if CPU_FAM_G0
rsource "Kconfig.clock"
endif
```
I tried locally and it works.

Reply:
> What do you think of the following (only include the clock config if using the corresponding family) ?
> 
> ```
> if CPU_FAM_G0
> rsource "Kconfig.clock"
> endif
> ```
> 
> I tried locally and it works.

Does this work for you together with changing `$(CPU_FAM)` to `*`?  When I make that change locally I get a lot of warnings. This might be due to the fact the Kconfig actually does not have conditional include, but just applies `CPU_FAM_Gx` as a dependency to all the symbols defined inside the `source`d file.

Reply:
The problem lies mainly on the choices, do you think it would be possible to factor the choices out?

Reply:
> Does this work for you together with changing $(CPU_FAM) to *?

Ah yes, that doesn't work! I messed up my local tests and thought it was OK. Sorry.

> Do you think it would be possible to factor the choices out?

Trying to anticipate this with other families and that might cause issues because not all families use the same choices options. Or maybe we could group all choices for all families and add them using conditionals on the corresponding families. The default choice can be set based on the family if there are special case. Is this what you are thinking about ?

Reply:
> Trying to anticipate this with other families and that might cause issues because not all families use the same choices options. Or maybe we could group all choices for all families and add them using conditionals on the corresponding families.

Would it work to have one choice and make the choice options to depend on the family symbols, or even on features to make it a bit more generic?

> The default choice can be set based on the family if there are special case. Is this what you are thinking about ?

Yes, we can use `.config` files to select the defaults here when needed.

Reply:
As a reference, would something like this apply to our case?

https://github.com/zephyrproject-rtos/zephyr/blob/88034a9dada32ee1da97c7a1e2029d56fb367b48/drivers/clock_control/Kconfig.stm32#L27-L57

Reply:
> As a reference, would something like this apply to our case?

That's basically what I was trying to say.

Reply:
> we can use .config files to select the defaults here when needed.

I think there's a confusion. I was not talking of the way to specify a specific config but how to define the available options.

Reply:
Reworked that PR following our discussion: all clock configurations are defined in a single `cpu/stm32/kconfigs/Kconfig.clk` file. This is much nicer and makes things simpler to integrate clock configurations of other families.